src/gpu/intel/ocl/micro_sdpa.cl

```diff
@@ -249,6 +249,49 @@ micro_sdpa(const global KEY_DATA_T *K, const global half *Q,
             / VAL_ZP_ELEMENTS_PER_BYTE;
 #endif
 
+#ifdef PREFETCH_K0
+    /* Prefetch first K tile. */
+    cooperative_prefetch_2d_k(
+            /* ptr */ K,
+            /* r */ k,
+            /* c */ d, // faster than D_MAX
+            /* rmax */ ugemm_kq_wg_tile_m,
+            /* cmax */ PREFETCH_D_MAX,
+            /* ld */ ldk,
+            /* sg_id */ sg_ij,
+            /* n_sg */ sg_per_wg,
+            /* sg_size */ SUBGROUP_SIZE,
+            /* cache */ LSC_LDCC_L1C_L3C);
+    //return;
+
+#if KEY_SCALES == QUANTIZE_2D
+    cooperative_prefetch_2d_maybe_rem(
+            /* ptr */ K_scales,
+            /* r */ k,
+            /* c */ D_MAX / KEY_GROUP_SIZE,
```

Comment:
Need to avoid out-of-bounds prefetches here, and similarly for the zp prefetches:
```suggestion
            /* c */ d / KEY_GROUP_SIZE,
```